airflow/utils/helpers.py

```diff
@@ -311,7 +312,14 @@ def exactly_one(*args) -> bool:
         raise ValueError(
             "Not supported for iterable args. Use `*` to unpack your iterable in the function call."
         )
-    return sum(map(bool, args)) == 1
+
+    def is_set(val):
+        if isinstance(val, ArgNotSet):
+            return False
+        else:
+            return bool(val)
+
+    return sum(map(is_set, args)) == 1
```

Comment:
```suggestion
    return sum(a is not NOT_SET and bool(a) for a in args) == 1
```

Does this read better?